static/controllers/favoritesController.js

```diff
@@ -1,94 +1,94 @@
-				
-				else {
-					$localStorage.favorite_stations.splice(index, 1);
-				}
-				
-				for (var i = 0; i < $scope.stations.length; i++) {
-					if ($scope.stations[i].id === station_id)
-						$scope.stations.splice(i, 1);
+
+			if (index >= 0) {
+				for (var i = 0; i < vm.stations.length; i++) {
+					if (vm.stations[i].id === station_id)
+						vm.stations.splice(i, 1);
 				}
-			
-			}
-		}
-		else {
-			$localStorage.favorite_stations = [station_id];			
+		    }
 		}
+
+		FavoritesService.addToFavorites(station_id);
 	};
-	
-	$scope.favoritesExist = FavoritesService.favoritesExist;
-	$scope.isFavorited = FavoritesService.isFavorited;
 
-	$scope.predictionText = function(station_id) {
+	vm.predictionText = function(station_id) {
 		var str = "";
 		
-		for (var i = 0; i<$scope.stations.length; i++) {
-			
-			var o_pre_1 = $scope.stations[i].outbound_pre.pre_1;
-			var o_pre_2 = $scope.stations[i].outbound_pre.pre_2;
-			var i_pre_1 = $scope.stations[i].inbound_pre.pre_1;
-			var i_pre_2 = $scope.stations[i].inbound_pre.pre_2;
-			
-			if ($scope.stations[i].id === station_id) {
-				if (o_pre_2 === null) {
-					if (o_pre_1 === null) {
-						 str = "Outbound: There is no scheduled service to this station at this time.<br />";
-					}
-					else {
-						str = "Next Service - Outbound: " + UtilityService.formatSeconds(o_pre_1) + "<br />";
-					}
-				}
-				else {
-				str = "Next Service - Outbound: " + UtilityService.formatSeconds(o_pre_1) + ", " + UtilityService.formatSeconds(o_pre_2) + "<br />";
-				}
-				
-				if (i_pre_2 === null) {
-					if (i_pre_1 === null) {
-						 str += "Next Service - Inbound: There is no scheduled service to this station at this time.";
-						break;
-					}
-					else {
-						str += "Next Service - Inbound: " + UtilityService.formatSeconds(i_pre_1);
-						break;
-					}
-				}
-				str += "Inbound: " + UtilityService.formatSeconds(i_pre_1) + ", " + UtilityService.formatSeconds(i_pre_2) + ".";
-				break;
+		for (var i = 0; i<vm.stations.length; i++) {
+			if (vm.stations[i].id === station_id) {
+                str += "Outbound: "
+                    + PredictionService.makePredictionString(vm.stations[i].outbound_pre.pre_1,
+                                                            vm.stations[i].outbound_pre.pre_2)
+                    + "<br />";
+                str += "Inbound: "
+                    + PredictionService.makePredictionString(vm.stations[i].inbound_pre.pre_1,
+                                                             vm.stations[i].inbound_pre.pre_2);
+				return str;
 			}
 		}
-		
-		return str;
 	}
+
+	var refreshFavoritesData = function() {
+	    for (var i = 0; i < vm.stations.length; i++) {
+            refreshOutboundPredictions(i);
+            refreshInboundPredictions(i);
+		}
+	};
+
+	var refreshInboundPredictions = function(station_idx) {
+	    $http.get('/station/' + vm.stations[station_idx].id + '/direction/' + 1 + '/nextservice')
```

Comment:
I have continued concerns over the number of http requests this involves but I'll be "that guy" in another ticket 👍  👻